tests/utils/profilehandler/profile_handler.go

```diff
@@ -565,3 +566,153 @@ func CreateClusterByProfile(profile *Profile, client *rosacli.Client, waitForClu
 
 	return description, err
 }
+
+func WaitForClusterUninstalled(client *rosacli.Client, cluster string, timeoutMin int) error {
+
+	endTime := time.Now().Add(time.Duration(timeoutMin) * time.Minute)
+	sleepTime := 0
+	for time.Now().Before(endTime) {
+		output, err := client.Cluster.DescribeClusterAndReflect(cluster)
+		if err != nil {
+			outputInfo, err := client.Cluster.DescribeCluster(cluster)
+			if strings.Contains(outputInfo.String(), "There is no cluster with identifier or name '"+cluster+"'") {
+				log.Logger.Infof("Cluster %s has been deleted.", cluster)
+				return nil
+			} else {
+				return err
+			}
+		}
+
+		switch output.State {
+		default:
+			if strings.Contains(output.State, CON.Uninstalling) {
+				time.Sleep(2 * time.Minute)
+				continue
+			}
+			if strings.Contains(output.State, CON.Ready) {
+				log.Logger.Infof("Cluster is in status of %v, wait for uninstalling", CON.Ready)
+				if sleepTime >= 6 {
+					return fmt.Errorf("cluster stuck to %s status for more than 6 mins.", output.State)
+				}
+				sleepTime += 2
+				time.Sleep(2 * time.Minute)
+				continue
+			}
+			if strings.Contains(output.State, CON.Error) {
+				log.Logger.Errorf("Cluster is in %s status now. Recording the installation log", CON.Error)
+				RecordClusterInstallationLog(client, cluster)
+				return fmt.Errorf("cluster %s is in %s state with reason: %s",
+					cluster, CON.Error, output.State)
+			}
+			return fmt.Errorf("unknown cluster state %s", output.State)
+
+		}
+	}
+	return fmt.Errorf("timeout for cluster ready waiting after %d mins", timeoutMin)
+}
+
+func DestroyClusterByProfile(client *rosacli.Client, waitForClusterUninstall bool) (errors []error) {
+	var (
+		ud                 UserData
+		cd                 ClusterDetail
+		clusterService     rosacli.ClusterService
+		errDeleteCluster   error
+		ocmResourceService rosacli.OCMResourceService
+	)
+
+	// get cluster info from cluster detail file
+	cdContent, err := common.ReadFileContent(config.Test.ClusterDetailFile)
+	if err != nil {
+		log.Logger.Errorf("Error happened when read cluster detail: %s", err.Error())
+		errors = append(errors, err)
+		return
+	}
+	fmt.Println(cdContent)
+	err = json.Unmarshal([]byte(cdContent), &cd)
+	if err != nil {
+		log.Logger.Errorf("Error happend when parse cluster detail file to ClusterDetail struct: %s", err.Error())
+		errors = append(errors, err)
+		return
+	}
+
+	// delete cluster
+	clusterService = client.Cluster
+	output, errDeleteCluster := clusterService.DeleteCluster(cd.ClusterID, "-y")
+	if errDeleteCluster != nil {
```

Comment:
If cluster deletion error, other resources cannot be deleted. You can just return here.

Reply:
And have you ever tried delete non-existing cluster, if it will fail? We should consider about cluster already being deleted status.